kratos/solving_strategies/builder_and_solvers/residualbased_block_builder_and_solver_with_constraints.h

```diff
@@ -602,11 +607,9 @@ class ResidualBasedBlockBuilderAndSolverWithConstraints
 
             if (constraint_is_active)
             {
-                it->EquationIdVector(slave_equation_ids, master_equation_ids, r_current_process_info);
-                //calculate constraint's T and b matrices
-                it->CalculateLocalSystem(relation_matrix, constant_vector, r_current_process_info);
                 //assemble the Constraint contribution
-                AssembleSlaves(slave_equation_ids, master_equation_ids, relation_matrix);
+                #pragma omp critical
```

Comment:
why critical? this kills the parallelism...

Reply:
the critical is necessary for synchronized access to the map `mGlobalMasterSlaveConstraints`, which is a map of id and a unique pointer. If there is no `critical` there, there will be random segfaults.